### PR TITLE
[EXPERIMENTAL] Make nil/true/false to_s return a frozen string

### PR DESCRIPTION
**Pull request by casperisfine**
Ref: https://bugs.ruby-lang.org/issues/16150?next_issue_id=16184
Ref: https://github.com/ruby/ruby/pull/2487
Ref: https://github.com/ruby/ruby/pull/2437

This is the last part of feature 16150.

cc @nobu @eregon

### CODE REVIEW

File: spec/ruby/core/true/to_s_spec.rb [hunk 1]
```diff
@@ -4,4 +4,10 @@
   it "returns the string 'true'" do
     true.to_s.should == "true"
   end
+
+  ruby_version_is "2.7" do
+    it "returns a frozen string ''" do
```
Comment (eregon) [inline spec/ruby/core/true/to_s_spec.rb]: `it "returns the frozen string 'true'" do`

File: spec/ruby/core/false/to_s_spec.rb [hunk 2]
```diff
@@ -4,4 +4,10 @@
   it "returns the string 'false'" do
     false.to_s.should == "false"
   end
+
+  ruby_version_is "2.7" do
+    it "returns a frozen string ''" do
```
Comment (eregon) [inline spec/ruby/core/false/to_s_spec.rb]: `it "returns the frozen string 'false'" do`

File: spec/ruby/core/false/to_s_spec.rb [hunk 3]
```diff
@@ -4,4 +4,10 @@
...
+      false.to_s.frozen?.should == true
+    end
```
Comment (eregon) [inline spec/ruby/core/false/to_s_spec.rb]: Could you add a spec that it's always the same String returned, as for Module#name & Symbol#to_s?

**Comment by eregon** [comment]
Looks good, @nobu can you review too?

**Comment by casperisfine** [comment]
@eregon sorry for the bad copy paste mistakes, I just fixed them.

I also added the same instance test. 

Also not sure why GItHub Actions won't build :/

File: object.c [hunk 4]
```diff
@@ -40,6 +40,10 @@ VALUE rb_cNilClass; /*!< NilClass class */
 VALUE rb_cTrueClass; /*!< TrueClass class */
 VALUE rb_cFalseClass; /*!< FalseClass class */
 
+VALUE rb_cNilClass_to_s;
+VALUE rb_cTrueClass_to_s;
+VALUE rb_cFalseClass_to_s;
```
Comment (nobu) [inline object.c]: Should be `static` if no specific reason.

Comment (casperisfine) [reply object.c]: Done.

**Comment by eregon** [comment]
I'm not sure why GitHub Actions don't trigger, maybe try amending the commit/rebase and push? cc @hsbt 
Could you maybe make sure `make test-all TESTS=-j8` and `make test-spec MSPECOPT=-j` pass locally?

**Comment by eregon** [comment]
I sent an email to GitHub Support since that's what https://github.com/ruby/ruby/pull/2490/checks?check_run_id=237695456 says to do.

**Comment by eregon** [comment]
@casperisfine it seems the issue with GitHub Actions not running is that the PR is from a fork of `github/ruby` instead of `ruby/ruby`.

**Comment by byroot** [comment]
@eregon ah! Not surprised. I meant to fix that a few weeks ago, but never did.

Do you want me to re-open another PR to get a proper CI run?

**Comment by eregon** [comment]
@byroot It's probably the easiest way, yes (sorry about that).
I'd like to see the CI run before merging :)

**Comment by byroot** [comment]
No worries, and not your fault.

**Comment by casperisfine** [comment]
Closing in favor of https://github.com/ruby/ruby/pull/2492

**Comment by eregon** [comment]
Here is what GitHub support replied:

> Thank you for reaching out to GitHub Support!
> 
> Our Engineers recently deployed a fix for this, could you give it another try and let me know if you encounter any other issues?

So, it should work next time :)

**Comment by byroot** [comment]
Thanks. I actually reforked directly from `ruby/ruby`, so that's fine. But it's a good thing they fixed it.

**Pull request closed without merging**